specification/metadata-3.2.md

```diff
@@ -45,151 +44,174 @@ of `METADATA LIST` MUST be as follows:
+case-insensitive. Values are unrestricted, except that they MUST be UTF-8.
+Server and client authors cannot assume that keys will carry specific
+formatting or meaning.
 
 If a limit is set for keys, it MUST only apply to user-set keys and MUST be
 communicated to the user via `RPL_ISUPPORT` as a numeric value on the
 `METADATA` key.
 
-### Predefined Keys
+## WHOIS
 
-The following keys are predefined for the purposes of the key registry outlined
-above:
+A subset of metadata MAY be sent via the `RPL_WHOISKEYVALUE` event; this
+subset SHALL be set explicitly, rather than informatively or as a side-effect
+of other events. For a complete view of user metadata, see `METADATA LIST`.
 
-| Key            | Meaning                                                      |
-| -------------- | ------------------------------------------------------------ |
-| server.certfp  | SSL certificate fingerprint                                  |
-| user.email     | User's email address                                         |
-| user.phone     | User's phone number                                          |
-| user.website   | User's website                                               |
-| user.im.*      | IM handles; the * is replaced with the relevant service name |
-| user.playing   | Music the user is currently listening to                     |
-| user.status    | The user's current status                                    |
-| client.name    | Client's name                                                |
-| client.version | Client version                                               |
+## IRC Daemons
 
-Except for `server.certfp`, all of these keys SHOULD be considered to be
-free-flowing text with no inherent meaning. `server.certfp` MUST be presented
-as hexadecimal.
+IRC servers may choose to accept or deny any key for any reason, and SHOULD
+implement a blacklist or whitelist functionality for this purpose, configurable
+by the server operators.
 
-## WHOIS
+If `METADATA` is supported, it MUST be specified in `RPL_ISUPPORT` using the
+`METADATA` key. Servers MAY specify a limit on the number of explicitly-set
+keys per-user; the format in that case MUST be `METADATA=<integer>`, where
+`<integer>` is the limit.
 
-Metadata MUST appear in `WHOIS` output using `RPL_WHOISKEYVALUE` if any is defined
-for the user.
+## Numerics
 
-## Services
+The numerics 760 through 769 MUST be reserved for metadata, carrying the
+following labels and formats:
 
-Metadata SHOULD be presentable via NickServ or similar service in order to
-allow for persistent metadata; the format for encapsulating `METADATA` events
-for persistence MUST be as follows:
+| No. | Label                 | Format                                     |
+| --- | --------------------- | ------------------------------------------ |
+| 760 | `RPL_WHOISKEYVALUE`   | `<Target> <Key> [<Visibility>] :<Value`    |
+| 761 | `RPL_KEYVALUE`        | `<Target> <Key>[ <Visibility>][ :<Value>]` |
```

Comment:
But how do you distinguish between

    :irc.example.com 760 user1 account :user1

which sets metadata without visibility, and

    :irc.example.com 760 user1 account 0

which clears metadata with visiblity?